components/builder-graph/src/graph_helpers.rs

```diff
@@ -443,23 +445,36 @@ pub fn transitive_deps(graph: &DiGraphMap<PackageIdentIntern, EdgeType>,
 // depend them, filtered by an origin 2) we find the packages in that set that are unbuildable
 // 3) we flood the graph to find all the packages that are rendered unbuildable because a dep is
 // unbuilable.
-pub fn compute_rebuild_set(graph: &DiGraphMap<PackageIdentIntern, EdgeType>,
-                           unbuildable: &dyn Unbuildable,
-                           touched: &[PackageIdentIntern],
-                           origin: &str,
-                           target: PackageTarget)
-                           -> Vec<PackageIdentIntern> {
+pub fn compute_rebuild_set(
+    graph: &DiGraphMap<PackageIdentIntern, EdgeType>,
+    unbuildable: &dyn Unbuildable,
+    touched: &[PackageIdentIntern],
+    origin: Option<&str>,
```

Comment:
It might be nice to eventually leverage [Origin](https://github.com/habitat-sh/habitat/blob/4fa20d0742bf0cb09d43f9695aef2b60893162cb/components/core/src/origin.rs#L9-L23) here.